heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -153,39 +153,41 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
       deactivateTopology(stateManager, topology, proposedPackingPlan);
     }
 
-    Set<PackingPlan.ContainerPlan> updatedContainers =
-        new HashSet<>(proposedPackingPlan.getContainers());
-    // request new resources if necessary. Once containers are allocated we should make the changes
-    // to state manager quickly, otherwise the scheduler might penalize for thrashing on start-up
-    if (newContainerCount > 0 && scalableScheduler.isPresent()) {
-      Set<PackingPlan.ContainerPlan> containersToAdd = containerDelta.getContainersToAdd();
-      Set<PackingPlan.ContainerPlan> containersAdded =
-          scalableScheduler.get().addContainers(containersToAdd);
-      // Update the PackingPlan with new container-ids
-      if (containersAdded != null) {
-        updatedContainers.removeAll(containersToAdd);
-        updatedContainers.addAll(containersAdded);
+    try {
+      Set<PackingPlan.ContainerPlan> updatedContainers =
+          new HashSet<>(proposedPackingPlan.getContainers());
+      // request new resources if necessary. Once containers are allocated we should make the changes
+      // to state manager quickly, otherwise the scheduler might penalize for thrashing on start-up
+      if (newContainerCount > 0 && scalableScheduler.isPresent()) {
+        Set<PackingPlan.ContainerPlan> containersToAdd = containerDelta.getContainersToAdd();
+        Set<PackingPlan.ContainerPlan> containersAdded =
+            scalableScheduler.get().addContainers(containersToAdd);
```

Comment:
Need to check whether we successfully add containers.

Reply:
updated